curp/src/server/curp_node.rs

```diff
@@ -320,22 +329,18 @@ impl<C: 'static + Command, RC: RoleChange + 'static> CurpNode<C, RC> {
     }
 
     /// Responsible for bringing up `sync_follower_task` when self becomes leader
-    async fn sync_follower_daemon(
-        curp: Arc<RawCurp<C, RC>>,
-        connect: Arc<impl ConnectApi + ?Sized>,
-        sync_event: Arc<Event>,
-    ) {
+    async fn sync_followers_daemon(curp: Arc<RawCurp<C, RC>>) {
         let leader_event = curp.leader_event();
         loop {
             if !curp.is_leader() {
                 leader_event.listen().await;
             }
-            Self::sync_follower_task(
-                Arc::clone(&curp),
-                Arc::clone(&connect),
-                Arc::clone(&sync_event),
-            )
-            .await;
+            let peers = curp.cluster().peers_addrs();
+            let futs = rpc::connect(peers).await.map(|(id, connect)| {
+                let sync_event = curp.sync_event(id);
+                Self::sync_follower_task(Arc::clone(&curp), connect, sync_event)
+            });
+            _ = futures::future::join_all(futs).await;
```

Comment:
Could spawn a tokio task for each `sync_follower_task` achieve better performance? As they could run on different thread.

Reply:
Fixed